src/upload/system/library/transbank/utils/LogHandler.php

```diff
@@ -0,0 +1,158 @@
+<?php
+namespace Transbank\Utils;
+use Monolog\Logger;
+use Monolog\Handler\RotatingFileHandler;
+
+class LogHandler
+{
+    //constants for log handler
+    const LOG_DEBUG_ENABLED = false; //enable or disable debug logs
+    const LOG_INFO_ENABLED = true; //enable or disable info logs
+    const LOG_ERROR_ENABLED = true; //enable or disable error logs
+
+    private $logDir;
+    private $logURL;
+    private $logger;
+
+    public function __construct($ecommerce = 'opencart')
+    {
+        $this->logDir = DIR_APPLICATION."logs/Transbank_webpay";
+        $this->logURL = str_replace($_SERVER['DOCUMENT_ROOT'], "", $this->logDir);
+        $this->setMakeLogDir();
+        $this->logger = new Logger('webpay_logger');
+        $this->logger->pushHandler(new RotatingFileHandler("{$this->logDir}/log_transbank_{$ecommerce}.log", 10, Logger::DEBUG));
+    }
+
+    /**
+     * Formats the size of a file in readable format.
+     * @param string $path
+     * @return string
+     */
+    private function formatBytes($path): string
+    {
+        $bytes = @filesize($path);
+        if ($bytes > 0) {
+            $unit = intval(log($bytes, 1024));
+            $units = array('B', 'KB', 'MB', 'GB');
+            if (array_key_exists($unit, $units) === true) {
+                return sprintf('%d %s', $bytes / pow(1024, $unit), $units[$unit]);
+            }
+        }
+        return '0 B';
+    }
+
+    /**
+     * Create the log directory if it does not exist.
+     */
+    private function setMakeLogDir(): void
+    {
+        try {
+            if (!file_exists($this->logDir)) {
+                mkdir($this->logDir, 0777, true);
+            }
+        } catch (\Exception $e) {
+            error_log('Error al crear la carpeta de logs ' . $e->getMessage());
```

Comment:
if create log folder fails, where is this error_log located?

Reply:
what is the answer? error_log is a php function and the message is handled by operating system
REF: https://www.php.net/manual/en/function.error-log.php

This is not the best way to handle it, you can show a message on the front end or try to write to the Opencart log